modules/lib/global.d.ts

```diff
@@ -193,11 +193,18 @@ declare const require: XpRequire;
  * var path = resolve('./other.html');
  *
  * @example
- * // Reslove absolute
+ * // Resolve absolute
  * var path = resolve('/path/to/other.html');
  *
  * @param {string} path Path to resolve.
  * @returns {*} Reference to an object.
  * @global
  */
-declare const resolve: (path: string) => object;
+declare const resolve: (path: string) => {
+    getApplicationKey(): string;
+    getPath(): string;
+    getUri(): string;
+    isRoot(): boolean;
+    getName(): string;
+    getExtension(): string;
+};
```

Comment:
Is there a reason why `ResourceKey` isn't used here? Like this:

```suggestion
declare const resolve: (path: string) => ResourceKey;
```

Reply:
Because ResourceKey isn't defined there. And as @edloidas said in his initial comment, he doesn't want to define it, nor import it from core.

Reply:
Sorry. I missed that. :) Maybe add a comment both in *core* and *global* to remind to update the other place, if the shape is changed.